src/exo/LoopIR_unification.py

```diff
@@ -1075,6 +1075,28 @@ def unify_types(self, pt, bt, pnode, bnode):
                 f"type {bt} (@{bnode.srcinfo})"
             )
 
+    @staticmethod
+    def comparision_to_unification_expr(e):
+        def sub(lhs, rhs):
+            return LoopIR.BinOp("-", lhs, rhs, T.index, null_srcinfo())
+
+        def add1(e):
+            one = LoopIR.Const(1, T.int, null_srcinfo())
+            return LoopIR.BinOp("+", e, one, T.index, null_srcinfo())
+
+        if e.op == "<" or e.op == "==":
+            # rewrite `lhs op rhs` into `0 op rhs - lhs`
+            return sub(e.rhs, e.lhs)
+        elif e.op == "<=":
+            # rewrite `lhs <= rhs` into `0 < rhs - lhs + 1`
+            return add1(sub(e.rhs, e.lhs))
+        elif e.op == ">":
+            # rewrite `lhs > rhs` into `0 < lhs - rhs`
+            return sub(e.lhs, e.rhs)
+        elif e.op == ">=":
+            # rewrite `lhs >= rhs` into `0 < lhs - rhs + 1`
+            return add1(sub(e.lhs, e.rhs))
```

Comment:
You probably want `else` here just for sanity

Reply:
Fixed.